sdk/go/pulumi-language-go/main.go

```diff
@@ -77,7 +82,17 @@ func findProgram(binary string) (*exec.Cmd, error) {
 		return nil, errors.Errorf("Failed to find go files for 'go run' matching %s", goFileSearchPattern)
 	}
 
-	return exec.Command(program, "run", cwd), nil
+	binaryToRun := path.Join(cwd, PULUMI_BINARY_NAME)
+	buildCmd := exec.Command(program, "build", "-o", binaryToRun, cwd)
+	if err := buildCmd.Run(); err != nil {
+		logging.V(5).Infof("Unable to `go build` falling back to `go run`")
+		// fallback to old behavior if failure. i.e.
+		// - compilation failure
```

Comment:
trying to limit the number of changes in this PR. `go run` will fail with its typical behavior on compilation failures.

Reply:
In what scenario could go run possibly work where go build doesn't? I think we can probably simplify and just remove it.